patches/tModLoader/Terraria/ModLoader/Default/TEModdedPylon.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Linq;
+using Terraria.DataStructures;
+using Terraria.ID;
+
+namespace Terraria.ModLoader.Default
+{
+	/// <summary>
+	/// This is a TML provided Tile Entity that acts extremely similar to vanilla's pylon TEs. If you plan
+	/// to make a pylon tile that closely resembles (or is identical to) vanilla's pylon TEs, you should use
+	/// this.
+	/// </summary>
+	public class TEModdedPylon : ModTileEntity, IPylonTileEntity
+	{
+
+		public override void NetPlaceEntityAttempt(int x, int y) {
```

Comment:
it would be nice if this could use more of the base impl. Also neither this nor `AdvancedPylonTileEntity` check `Main.PylonSystem.HasPylonOfType` like `TETeleportationPylon` does